File: Mouse.js
class Mouse {
  constructor(position={x:0,y:0}, velocity={x:1,y:0}, width=32, height=32) {
    this.position = position;
    this.velocity = velocity;
    this.width = width;
    this.height = height;
    this.onGround = false;
    this.speed = 2;
  }

  draw() {
  }

  move() {
    if (this.controls.left) {
      this.velocity.x = -this.speed;
    } else if (this.controls.right) {
      this.velocity.x = this.speed;
    } else {
      this.velocity.x = 0;
    }
    if (this.controls.jump) {
      this.jump();
    }
    let newPos = {
      x:this.position.x+this.velocity.x,
      y:this.position.y+this.velocity.y
    };
    this.position = newPos; 
    
    if (!this.onGround) {
      this.velocity.y += 0.5;
      if (isSolid(newPos.x, newPos.y)) {
        this.velocity.y = 0;
        this.onGround = true;
      }
    } else {
      if (!isSolid(this.position.x, this.position.y+1)) {
        this.onGround = false;
      }
    }
    while(isSolid(this.position.x, this.position.y)) {
      this.position.y -= 1;
      this.onGround = true;
    }
  }

  jump() {
    if (this.onGround) {
      this.velocity.y = -10;
      this.onGround = false;
    }
  }

  climb() {

  }

  teleport(position = this.position) {
    this.position = position;
  }

  setVelocity(velocity=this.velocity) {
    this.velocity = velocity;
    return this.velocity;
  }

  update() {
    this.move();
  }

}

File: mouse.py
import pygame, sys
from pytmx import load_pygame

pygame.init()
size = width, height = 800, 600

screen = pygame.display.set_mode(size)

tilesize = 32

class Mouse:
    def __init__(self):
        self.x = 0
        self.y = 0
        self.climbing = False
        self.touchingGround = True

class Level:
    def __init__(self, path):
        self.tmxdata = load_pygame(path)

    def draw(self, camera, screen):
        for i in range(100):
            for j in range(100):      
                position = (camera.x+i*tilesize, camera.y+j*tilesize)
                image = self.tmxdata.get_tile_image(
                    camera.x//tilesize+i, 
                    camera.y//tilesize+j,
                    0
                )
                screen.blit(image, position)


class Camera:
    def __init__(self, mouse):
        self.mouse = mouse
        self.x = mouse.x
        self.y = mouse.y
        self.widthtiles = 20
        self.heighttiles= 20

currentlevel = Level("testlevel.tmx")
mouse = Mouse()
camera = Camera(mouse)

while True:
    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            sys.exit()

    currentlevel.draw(camera, screen)
    screen.flip()
    pygame.display.update()
    pygame.time.delay(1000/60)

    

    